src/Adapter/Order/OrderProductQuantityUpdater.php

```diff
@@ -391,116 +385,29 @@ protected function reinjectQuantity(
         $reinjectableQuantity = $oldQuantity - $newQuantity;
         $quantityToReinject = $oldQuantity > $reinjectableQuantity ? $reinjectableQuantity : $oldQuantity;
 
-        $product = new Product(
+        StockAvailable::updateQuantity(
```

Comment:
Are you sure?

Reply:
This is the code that was executed when `id_warehouse == 0` (it's always zero)